api_core/google/api_core/client_options.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Client options class.
+
+Client options provide an consistent interface for user options to be defined
+across clients.
+"""
+
+
+class ClientOptions(object):
+    """Client Options used to set options on clients. 
+
+    Args:
+        options (dict): A dict of the options listed below.
+        api_endpoint (str): The desired API endpoint, e.g., compute.googleapis.com
+    """
+
+    api_endpoint = None
+
+    def __init__(self, options=None, **kw_args):
+        if options is not None:
+            if kw_args.items():
+                raise Exception(
+                    "ClientOptions expects options in a dictionary or in kw_args, not both"
```

Comment:
I had some trouble wordsmithing here. Suggestions would be appreciated.

Reply:
It's a new class:  why don't we just use named arguments, so we can document them properly?  We don't have any backward-compatibility reason to deal with arbitrary arguments here.

Reply:
+1 I think we can just use the 'options' here